olive/platform_sdk/qualcomm/runner.py

```diff
@@ -43,32 +44,37 @@ def __init__(
     def _use_olive_env(self):
         return os.environ.get(USE_OLIVE_ENV, USE_OLIVE_ENV_DEFAULT_VALUE) == USE_OLIVE_ENV_DEFAULT_VALUE
 
-    def _resolve_cmd(self, cmd: str):
+    def _resolve_cmd(self, cmd: Union[str, List[str]]) -> List[str]:
+        # TODO(trajep): use list instead of string to avoid shlex.split error in non-posix mode
         import platform
 
-        if platform.system() == "Windows" and cmd.startswith(("snpe-", "qnn-")):
-            logger.debug("Resolving command %s on Windows.", cmd)
+        if isinstance(cmd, str):
+            cmd_list = shlex.split(cmd, posix=(platform.system() != "Windows"))
+        else:
+            cmd_list = cmd
+
+        if platform.system() == "Windows" and cmd_list[0].startswith(("snpe-", "qnn-")):
+            logger.debug("Resolving command %s on Windows.", cmd_list)
             cmd_dir = Path(self.sdk_env.sdk_root_path) / "bin" / self.sdk_env.target_arch
-            cmd_name = cmd.split(" ")[0]
+            cmd_name = cmd_list[0]
             cmd_full_path = cmd_dir / cmd_name
             if cmd_full_path.with_suffix(".exe").exists():
                 cmd_full_path = cmd_full_path.with_suffix(".exe")
             if cmd_full_path.exists():
-                cmd = str(cmd_full_path) + cmd[len(cmd_name) :]
+                cmd_list.insert(0, str(cmd_full_path))
```

Comment:
This needs to be cmd_list[0]=... instead of insert. We are directly replacing the first command with full path

Reply:
good catch